src/Antiforgery/src/AntiforgeryMiddleware.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Antiforgery.Infrastructure;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Antiforgery.Internal;
+
+internal sealed class AntiforgeryMiddleware(IAntiforgery antiforgery, RequestDelegate next)
+{
+    private readonly RequestDelegate _next = next;
+    private readonly IAntiforgery _antiforgery = antiforgery;
+
+    private const string AntiforgeryMiddlewareWithEndpointInvokedKey = "__AntiforgeryMiddlewareWithEndpointInvoked";
+    private static readonly object AntiforgeryMiddlewareWithEndpointInvokedValue = new object();
+
+    public async Task Invoke(HttpContext context)
+    {
+        var endpoint = context.GetEndpoint();
+
+        if (endpoint is not null)
+        {
+            context.Items[AntiforgeryMiddlewareWithEndpointInvokedKey] = AntiforgeryMiddlewareWithEndpointInvokedValue;
+        }
+
+        var method = context.Request.Method;
+        if (HttpMethods.IsGet(method) ||
+            HttpMethods.IsHead(method) ||
+            HttpMethods.IsTrace(method) ||
+            HttpMethods.IsOptions(method))
+        {
+            await _next(context);
+            return;
+        }
+
+        if (endpoint is not null &&
+            endpoint.Metadata.GetMetadata<IAntiforgeryMetadata>() is {} antiforgeryMetadata &&
+            antiforgeryMetadata.Required)
```

Comment:
Could you simplify this down to this one-liner (I didn't check it compiles)?

```suggestion
        if (endpoint?.Metadata.GetMetadata<IAntiforgeryMetadata>() is { Required: true })
```